Sources/PackageRegistryTool/PackageRegistryTool+Auth.swift

```diff
@@ -23,26 +23,26 @@ import struct TSCBasic.SHA256
 #if os(Windows)
 import WinSDK
 
-private func getpass(_ prompt: String) -> UnsafePointer<CChar> {
+private func readpassword(_ prompt: String) throws -> String {
     enum StaticStorage {
         static var buffer: UnsafeMutableBufferPointer<CChar> =
-            .allocate(capacity: 255)
+            .allocate(capacity: SwiftPackageRegistryTool.Login.passwordBufferSize)
     }
 
     let hStdIn: HANDLE = GetStdHandle(STD_INPUT_HANDLE)
     if hStdIn == INVALID_HANDLE_VALUE {
-        return UnsafePointer<CChar>(StaticStorage.buffer.baseAddress!)
+        throw StringError("unable to read input")
```

Comment:
Could these error strings give more information at least for debugging purposes? Maybe they should mention the function call that failed? If we receive a bug report that mentions this string error, it will be harder to pinpoint at which exact line it failed if all strings are the same.

Reply:
https://github.com/apple/swift-package-manager/pull/6562/commits/072ea716ba47fd719b587b9bccb061f897e59472